plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/ElasticsearchMetadata.java

```diff
@@ -269,16 +318,16 @@ private static boolean supportsPredicates(IndexMetadata.Type type)
 
     private Type toTrinoType(IndexMetadata.Field metaDataField)
     {
-        return toTrinoType(metaDataField, metaDataField.isArray());
-    }
-
-    private Type toTrinoType(IndexMetadata.Field metaDataField, boolean isArray)
-    {
+        checkArgument(!metaDataField.asRawJson() || !metaDataField.isArray(), format("Single column, (%s) cannot be declared as a Trino array and also be cast as raw json.",
```

Comment:
IIRC there was a comment from Martin saying that we should say something like "also be rendered as json" in the error messages

Reply:
good catch! I swear I had made that before...I may have lost it while applying fixes...there are two locations where we check for this so i'm fixing both.